### PR TITLE
Add component type and publish setup

### PR DESCRIPTION
**Pull request by margaree**
Fixes https://github.com/BrightspaceUI/lit-element-template/issues/5 and https://github.com/BrightspaceUI/lit-element-template/issues/3

### CODE REVIEW

File: configure-repo.js [hunk 1]
```diff
@@ -1,13 +1,42 @@
...
-	'packageName': '',
+	'codeowner': '',
 	'description': '',
-	'codeowner': ''
+	'publish': '',
+	'shortName': '',
+	'type': ''
 };
 
+if (config.type !== 'labs' && config.type !== 'official') {
```
Comment (margaree) [inline configure-repo.js]: I set this so type had to be either labs or official. Do we want to enforce this or should this be more open?

Comment (dlockhart) [reply configure-repo.js]: I like restricting it for now.

**Comment by dlockhart** [comment]
There's also some [stuff that goes in .travis.yml](https://github.com/BrightspaceUI/core/blob/master/.travis.yml#L34) for publishing to NPM. They'll need to manually add an encrypted key that they can request from us (the d2ltravisdeploy user).

Core has that auto-versioning code too that automatically updates the version in `package.json`... although we could leave that out for now.

**Comment by margaree** [comment]
Right I forgot about the travis things - I'll add those.
I had been planning to add the auto-versioning separately (https://github.com/BrightspaceUI/lit-element-template/issues/6)

**Pull request merged**